### DISCUSSION TITLE
How to manually navigate to no-found.tsx page outside of components, with static export?

### DISCUSSION THREAD

**Discussion by alienzhangyw** [vercel/next.js#51166] [Help] [answered]
### Summary

I want to redirect to the custom 404 page inside axios interceptors. Both redirect() and notFound() are not working.
```typescript
axios.interceptors.response.use((response) => {
  const { data } = response;
  if (data.code === 200) {
    return data.data;
  } else {
    if (data.code === 404) {
      // redirect('/404'); not working
      // notFound(); not working
      // window.location.href = '/404'; redirects to nextjs default 404.html
    } else return Promise.reject(data.msg);
  }
}, (error) => {
  return Promise.reject(error);
});
```

**Comment by devjiwonchoi** [comment]
Shouldn't you use `status` instead of `code`?

```js
instance.interceptors.response.use(
  (response) => {
    if (response.status === 404) {
      window.location.href = '/404'
    }
    return response
  },
  (error) => {
    return Promise.reject(error)
  }
)
```

Note that `/404` is not your default `404.html`, but is considered as `app/404/page.js`.

**Reply by alienzhangyw** [reply]
No, we use `status === 200` for all responses, and have a code within the response data, for example:
```json
{
    "code": 200,
    "msg": "success",
    "data": "some data"
}
```
And `code === 404` means resource not found.

**Reply by devjiwonchoi** [reply]
If so, the object destructuring should be `{ data, code } = response` and use `code` instead of `data.code`.

## Reference
- Object Destructuring

**Reply by alienzhangyw** [reply]
No, `response.data` is actually your JSON data in axios `Response`.
https://www.axios-http.cn/en/docs/res_schema

**Reply by devjiwonchoi** [reply, marked as answer]
Yeah, you're right my bad. In order to catch inside the `notFound` boundary, it needs to be in the route segment, in which, the Middleware nor `Interceptor` will successfully call `notFound`.

> The notFound function allows you to render the [not-found file](https://nextjs.org/docs/app/api-reference/file-conventions/not-found) within a route segment...

This [issue](https://github.com/vercel/next.js/issues/38239) has been brought up before but is not yet improved.

The most used solution for now is to create a custom `/404` and catch `notFound` in that route.

```js
// app/404/page.tsx

import { notFound } from 'next/navigation'
export default function CustomNotFound() {
  notFound()
}
```

By doing this, you can `window.location.href = '/404' and render your `not-found` page.

## Reference
- #38239
